msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -32,29 +34,29 @@
 import io.github.msdk.datamodel.rawdata.MsScanType;
 import io.github.msdk.datamodel.rawdata.PolarityType;
 import io.github.msdk.datamodel.rawdata.RawDataFile;
+import io.github.msdk.spectra.spectrumtypedetection.SpectrumTypeDetectionAlgorithm;
+import io.github.msdk.util.MsSpectrumUtil;
 import io.github.msdk.util.tolerances.MzTolerance;
 import it.unimi.dsi.io.ByteBufferInputStream;
 
 public class MzMLSpectrum implements MsScan {
-  private HashMap<String, String> cvParamValues;
+  private ArrayList<MzMLCVParam> cvParams;
   private MzMLBinaryDataInfo mzBinaryDataInfo;
   private MzMLBinaryDataInfo intensityBinaryDataInfo;
   private ByteBufferInputStream mappedByteBufferInputStream;
-
-  public MzMLSpectrum() {
-    cvParamValues = new HashMap<>();
-  }
-
-  public void add(String accession, String value) {
-    cvParamValues.put(accession, value);
+  private String id;
+  private Integer scanNumber;
+  private double[] mzValues;
```

Comment:
You cannot hold mzValues[] and intensityValues[] as variables, that takes too much memory. The whole point is to decode them on-demand.